src/Join_SQL.php

```diff
@@ -36,6 +36,12 @@ public function getDesiredName()
      */
     public function getDSQLExpression($q)
     {
+        // If our Model has expr() method (inherited from Persistence_SQL) then use it
+        if ($this->owner->hasMethod('expr')) {
+            return $this->owner->expr('{}.{}', [$this->foreign_alias, $this->foreign_field]);
+        }
```

Comment:
join_sql shouldn't even be called if expr is undefined. I think we should leave it here to produce error.